### PR TITLE
Fix detection of Optional type with a None value

### PR DESCRIPTION
**Pull request by daniel-thom**
Fixes a bug that was causing failures in the registration of dGen, which was mapping some model years to null.

### CODE REVIEW

File: dsgrid/utils/spark.py [hunk 1]
```diff
@@ -342,6 +342,33 @@ def models_to_dataframe(models, table_name=None):
...
+        if not types:
+            msg = f"Unhandled Union type: {python_type=} {args=}"
+            raise NotImplementedError(msg)
+        elif len(types) > 1:
+            msg = f"Unhandled Union type: {types=}"
+            raise NotImplementedError(msg)
+        else:
+            python_type = types[0]
+
+    return python_type
```
Comment (elainethale) [inline dsgrid/utils/spark.py:369]: Are you trying to find the type that this field is when it's not None, so you can set the type of the Spark dataframe column correctly?

File: tests/test_spark_utils.py [hunk 2]
```diff
@@ -18,3 +21,8 @@ def test_try_read_dataframe_valid(tmp_path):
     df = try_read_dataframe(filename)
     assert isinstance(df, DataFrame)
     assert df.collect()[0].a == 1
+
+
+def test_get_type_from_union():
+    assert get_type_from_union(Optional[str]) is str
+    assert get_type_from_union(Optional[DayType]) is str
```
Comment (elainethale) [inline tests/test_spark_utils.py:28]: So Optional[X] is the same as Union[X,None]?

File: dsgrid/utils/spark.py (hunk 1 above)
Comment (daniel-thom) [reply dsgrid/utils/spark.py:369]: yes

File: tests/test_spark_utils.py (hunk 2 above)
Comment (daniel-thom) [reply tests/test_spark_utils.py:28]: yes

**Review by ashreeta — APPROVED** [review]

**Pull request merged**